README.md

```diff
@@ -445,9 +445,120 @@ To build dense indexes (e.g., Faiss indexes) on your own document collections, f
 <details>
 <summary>Guide to indexing and searching English documents</summary>
 
-TODO: Clean up and merge document from the following pages:
-+ [Dense indexes](docs/usage-dense-indexes.md)
-+ [Encode](docs/usage-encode.md)
+To build the dense index, Pyserini allows to either directly build Faiss Flat index via `pyserini.encode` with `output --to-faiss`, 
+or first encode collections into vectors via `pyserini.encode`, then build various types of Faiss index via `pyserini.index.faiss` based on the encoded collections. 
+ 
+To use the `pyserini.encode`, the input should be in JSONL format. 
+Each line is a json dictionary containing two fields, i.e .`id` and `contents`.
+- `id` is the document id in string.
+- `contents` contains all the fields of the documents. By default, Pyserini expects the fields in contents are separated by `\n`. The field's boundary can be controled using `--delimiter` argument under `input`, see the example script below.
+
+```json
+{
+  "id": "doc1",
+  "contents": "www.url.com\ntitle\nthis is the contents.\ndocument expansion"
+}
+```
+In the above example, the document has four fields in contents, `url`, `title`, `text` and `expand`.
+
+With the collection in the correct foramt, we can now encode documents with Dense encoders:
+```bash
+python -m pyserini.encode \
+  input   --corpus msmarco-passage-expanded \
+          --fields url title text expand \  # fields in collection contents
+          --delimiter "\n" \
+          --shard-id 0 \   # The id of current shard. Default is 0
+          --shard-num 1 \  # The total number of shards. Default is 1
+  output  --embeddings path/to/output/dir \
+          --to-faiss \
+  encoder --encoder castorini/tct_colbert-v2-hnp-msmarco \
+          --fields title text \  # fields to encode
+          --batch 32 \
+          --fp16  # if inference with autocast()
+```
+* with `--to-faiss`, the generated embeddings will be stored as FaissIndexIP directly.  Otherwise it will be stored in `.jsonl` format.  If in `.jsonl` format, each line contains following info:
+```json
+{
+  "id": "doc1",
+  "contents": "www.url.com\ntitle\nthis is the contents.\ndocument expansion",
+  "vector": [0.12, 0.12, 0.13, 0.14]
+}
+```
+* The `shard-id` and `shard-num` arguments are for speeding up the encoding, where the `shard-num` controls the total shard you want to segment the collection into, and the `shard-id` is the id of the current shard to encode. For example, if `shard-num` is 4 and `shard-id` is 0, the command would create a sub-index for the first 1/4 of the collection. Then you can run 4 process on 4 gpu to speed up the process by 4 times.  Once it's done, you can merge the sub-indexes together by:
+```bash
+python -m pyserini.index.merge_faiss_indexes --prefix indexes/dindex-sample-dpr-multi- --shard-num 4
+```
+
+#### Encode documents with Sparse encoder
```

Comment:
Not sure how should I blend this section into the current "Sparse Indexes" docs?

Reply:
hrm. Let's leave in for now, I'll take another pass when I have time...